### DISCUSSION TITLE
`<C-Left/Right/Up/Down>` keymaps doesn't work when running in tmux

### DISCUSSION THREAD

**Discussion by gengdz** [LazyVim/LazyVim#322] [Q&A]
I know this is not a lazyVim problem, but if anyone else is experiencing this problem, please help me.

useful info:
* `<A-Left/Right/Up/Down>` can work well
* It's ok if i don't run in tmux

.tmux.conf
```bash
set-option -g status-position bottom

setw -g aggressive-resize on

set-option -sg escape-time 10
set-option -g focus-events on

set -g default-terminal "tmux-256color"
set-option -a terminal-overrides ",*256col*:RGB"

set -g default-shell $SHELL

set-window-option -g automatic-rename on
set-option -g set-titles on
 
# set -g status-keys vi
set -g history-limit 10000

# set -g mode-keys vi

set -g mouse on


bind h select-pane -L
bind j select-pane -D
bind k select-pane -U
bind l select-pane -R

# start window numbers at 1 to match keyboard order with tmux order
set -g base-index 1

# start pane numbers at 1 to match keyboard order with tmux order
setw -g pane-base-index 1

# easy resizing of panes
# bind -r J resize-pane -D 5
# bind -r K resize-pane -U 5
# bind -r H resize-pane -L 5
# bind -r L resize-pane -R 5

# bind-key -n C-Left  "resize-pane -L 5"
# bind-key -n C-Right "resize-pane -R 5"
# bind-key -n C-Up    "resize-pane -U 5"
# bind-key -n C-Down  "resize-pane -D 5"
 
# for vim
set -g @resurrect-strategy-vim 'session'
  # for neovim
set -g @resurrect-strategy-nvim 'session'

# plugins
source "~/.config/tmux/plugins.conf"

```

tmux plugins.conf
```bash
# plugins install start
# type prefix + I to fetch plugins
# List of plugins
set -g @plugin 'tmux-plugins/tpm'
set -g @plugin "arcticicestudio/nord-tmux"
set -g @plugin 'tmux-plugins/tmux-resurrect'
set -g status-keys vi

# Initialize TMUX plugin manager (keep this line at the very bottom of tmux.conf)
run -b '~/.config/tmux/tpm/tpm'


```

**Comment by ZerGo0** [comment]
Did you check your `.tmux` file already? Sounds like they could be set there.

**Reply by gengdz** [reply]
I updated the configuration file in the description.

**Reply by ZerGo0** [reply]
Try to use  `showkey` or `xev` to see if and what keys are actually being pressed. The config seems fine as far as I can see.

Is is possible that the keymaps are in `/etc/tmux.conf`, but I highly doubt it.

**Comment by ooloth** [comment]
Do you still notice the issue if you comment out these lines and reload your tmux config?

```
bind h select-pane -L
bind j select-pane -D
bind k select-pane -U
bind l select-pane -R
```

If so, I’m afraid I’m not sure what’s wrong. But I can confirm all LazyVim keymaps are working for me inside tmux. 

FWIW, to get the window movements you’re looking for, I’d recommend installing [vim-tmux-navigator](https://github.com/christoomey/vim-tmux-navigator) (in both tmux and LazyVim) and then adding [these](https://github.com/LazyVim/LazyVim/discussions/277#discussioncomment-5004092) keymaps to your `lua/config/keymaps.lua`. You’ll be able to use `<C-hjkl>` to move between both vim and tmux splits.

**Comment by gengdz** [comment]
Im so sorry,I got the title wrong!

**Reply by ZerGo0** [reply]
> Im so sorry,I got the title wrong!

Try setting the following values: 
```
set-option -g default-terminal "xterm-256color"
set-window-option -g xterm-keys on
```

**Reply by ooloth** [reply]
> Im so sorry,I got the title wrong!

Ah, I see. In that case, I have noticed the same problem!

**Comment by manishprivet** [comment]
Hi, for anyone looking to fix this issue, I solved it by adding this into my `keymaps.lua` file

```lua
vim.keymap.set("n", "<C-h>", ":<C-U>TmuxNavigateLeft<cr>", { noremap = true, silent = true })
vim.keymap.set("n", "<C-l>", ":<C-U>TmuxNavigateRight<cr>", { noremap = true, silent = true })
vim.keymap.set("n", "<C-j>", ":<C-U>TmuxNavigateDown<cr>", { noremap = true, silent = true })
vim.keymap.set("n", "<C-k>", ":<C-U>TmuxNavigateUp<cr>", { noremap = true, silent = true })
vim.keymap.set("n", "<C-\\>", ":<C-U>TmuxNavigatePrevious<cr>", { noremap = true, silent = true })
```

Works seamlessly
